File: c++/2149.cpp
// Link: https://leetcode.com/problems/rearrange-array-elements-by-sign/description/
// Runtime: 131 ms
// Memory: 126.4 MB

class Solution {
public:
  std::vector<int> rearrangeArray(const std::vector<int> &nums) {
    std::vector<int> rearrangedArray(nums.size());
    int positiveIndex = 0, negativeIndex = 1;
    for (const auto &val : nums) {
      if (val > 0) {
        rearrangedArray[positiveIndex] = val;
        positiveIndex += 2;
      } else {
        rearrangedArray[negativeIndex] = val;
        negativeIndex += 2;
      }
    }

    return rearrangedArray;
  }
};

File: python/2149.py
# Link: https://leetcode.com/problems/rearrange-array-elements-by-sign/description/
# Runtime: 1010 ms
# Memory: 47.19 MB


class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        rearranged_array = [0] * len(nums)
        positive_index, negative_index = 0, 1
        for val in nums:
            if val > 0:
                rearranged_array[positive_index] = val
                positive_index += 2
            else:
                rearranged_array[negative_index] = val
                negative_index += 2

        return rearranged_array
